### PR TITLE
Test with a time zone

### PR DESCRIPTION
**Pull request by pawelkaczor**

### CODE REVIEW

File: modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    assertEquals(refInUtc.getHour, 10)
+    assertEquals(refInUtc.getZone, ZoneId.of("UTC"))
+
+    val ce = CalEvent(Mon ~ Tue, DateEvent.All, time(11.c, 23.c, 0.c))
+
+    val result: Option[ZonedDateTime] = ce.nextElapse(refInUtc)
+
+    assertEquals(result.map(_.getZone), Some(ZoneId.of("UTC")))
+    assertEquals(result, Some(refInUtc.withMinute(23)))
```
Comment (pawelkaczor) [inline modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: => Diff (- obtained, + expected)
 Some(
-  value = 2021-11-15T11:23Z[UTC]
+  value = 2021-11-15T10:23Z[UTC]
 )

Comment (eikek) [reply modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: Hi @pawelkaczor sorry for the delay - finally looked at it.

I think your test is a bit off. The `refInUtc` is same as:
```scala
val refInUtc: ZonedDateTime =
    ZonedDateTime.of(2021, 11, 15, 10, 15, 0, 0, ZoneOffset.UTC)
```
It just specifies some date in UTC, the CET timezone is never observed after line 14. Then the calendar event is specified without a timezone, which means the one from the reference date is taken. The result is correctly `11:23 UTC`.

But the bug existed anyways :-) I think/hope I fixed it in #250

Comment (pawelkaczor) [reply modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: Hi @eikek,
your refInUtc is not really UTC date time (its time zone is Z) so it is not an equivalent of my refInUtc.

Your refInUtc:  
```scala
val refInUtc: ZonedDateTime =
    ZonedDateTime.of(2021, 11, 15, 10, 15, 0, 0, ZoneOffset.UTC)
```

My refInUtc (without the convertion from CET) :

```scala
    val refInUtc: ZonedDateTime =
      ZonedDateTime.of(2021, 11, 15, 10, 15, 0, 0, ZoneId.of("UTC"))
```

Comment (pawelkaczor) [reply modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: >> But the bug existed anyways :-) I think/hope I fixed it in https://github.com/eikek/calev/pull/250

I merged your changes (from master) but the test still fails.

Comment (eikek) [reply modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: Hi @pawelkaczor 

The `Z` is in some sense equivalent to UTC. It is the zero offset to UTC. Both dates are the same point in time. Only the timezone _id_ is different.

I would still think your test is not testing the bug and is failing correctly. The `refInUtc` is in UTC and the calendar event is in UTC, so the result of `nextElapse` will also be in UTC. The ref is `10:15:00 UTC`, the event `*-*-* 11:23:00` and so the result is correctly `11:23:00 UTC`, no? Why do you think it should be `10:23:00 UTC`? Or I'm not understanding what you mean, maybe.

Comment (pawelkaczor) [reply modules/core/src/test/scala/com/github/eikek/calev/CalEventTimeZoneTest.scala:24]: Perhaps I overlooked a bug in my test. I just changed the calendar event to: CalEvent(Mon ~ Tue, DateEvent.All, time(All, 23.c, 0.c)) and the test works as expected.

**Pull request closed without merging**